module-bluetooth/Bluetooth/interface/profiles/HFP/HFP.cpp

```diff
@@ -276,7 +276,7 @@ namespace bluetooth
             scoHandle = HCI_CON_HANDLE_INVALID;
             isAudioRouted                = false;
             isAudioConnectionEstablished = false;
-            sendAudioEvent(audio::EventType::BlutoothHFPDeviceState, audio::Event::DeviceState::Disconnected);
+            audioDevice.reset();
```

Comment:
can't it be somehow handled better in module-audio?

Reply:
but why in BT thread?
there is no race?

Reply:
Module audio doesn't get messages from bt stack. Sure it could be done in module audio, but to my mind we would need to have another message from module bt to module audio - that approach is error prone cause of the extra time that is needed to deliver a message plus we would need to handle a case in which message can't be delivered for whatever reason.

Reply:
@alekrudnik audiodevice is part of hfp pimpl

Reply:
sorry but I'm not convinced. `would need to handle a case in which message can't be delivered for whatever reason.` It looks like a very poor design here. Do we have any documentation for that?